x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/random/RandomRankBuilder.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.rank.random;
+
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankBuilder;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.search.rank.context.QueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.QueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankShardContext;
+import org.elasticsearch.search.rank.feature.RankFeatureDoc;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingRankFeaturePhaseRankShardContext;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+import static org.elasticsearch.xpack.inference.rank.random.RandomRankRetrieverBuilder.FIELD_FIELD;
+import static org.elasticsearch.xpack.inference.rank.random.RandomRankRetrieverBuilder.MIN_SCORE_FIELD;
+
+/**
+ * A {@code RankBuilder} that performs reranking with random scores, used for testing.
+ */
+public class RandomRankBuilder extends RankBuilder {
+
+    public static final String NAME = "random_reranker";
+
+    static final ConstructingObjectParser<RandomRankBuilder, Void> PARSER = new ConstructingObjectParser<>(NAME, args -> {
+        Integer rankWindowSize = args[0] == null ? DEFAULT_RANK_WINDOW_SIZE : (Integer) args[0];
+        String field = (String) args[1];
+        Float minScore = (Float) args[2];
+
+        return new RandomRankBuilder(rankWindowSize, field, minScore);
+    });
+
+    static {
+        PARSER.declareInt(optionalConstructorArg(), RANK_WINDOW_SIZE_FIELD);
+        PARSER.declareString(constructorArg(), FIELD_FIELD);
```

Comment:
There's a disconnect with [RandomRankRetrieverBuilder](https://github.com/elastic/elasticsearch/pull/111851/files#diff-1b108da71ea6470e309ce539402bad33fe0315049a63a23b9f59da83614688b6R51), in which `field` is mandatory - is that intentional?

I think it's OK to make `field` optional throughout this implementation, unless some parent process requires it to be present. Even better if we can remove it entirely, since it doesn't participate in the actual reranking (this might not be possible due to the rank feature phase's logic).